app/javascript/components/workflows/professor/exams/new/editor/components/BodyItem.tsx

```diff
@@ -1,60 +1,99 @@
 import { BodyItem } from '@student/exams/show/types';
 import { ExhaustiveSwitchError } from '@hourglass/common/helpers';
-import { MoveBodyItemAction, DeleteBodyItemAction } from '@professor/exams/new/types';
-import { moveBodyItem, deleteBodyItem } from '@professor/exams/new/actions';
-import MoveItem from '@professor/exams/new/editor/containers/MoveItem';
 import { Card } from 'react-bootstrap';
+import { Field, WrappedFieldProps, FormSection } from 'redux-form';
+import MoveItem from '@professor/exams/new/editor/components/MoveItem';
+import { EditHTMLField } from './editHTMLs';
 
-
-export interface BodyProps {
-  body: BodyItem;
+const OneItem: React.FC<WrappedFieldProps & {
+  memberName: string;
   qnum: number;
   pnum: number;
   bnum: number;
-  numBodyItems: number;
-}
-
-const Body: React.FC<BodyProps> = (props) => {
+}> = (props) => {
   const {
-    body, qnum, pnum, bnum,
+    input,
+    memberName,
+    qnum,
+    pnum,
+    bnum,
   } = props;
-  switch (body.type) {
-    case 'HTML':
-      return <EditHTML info={body} qnum={qnum} pnum={pnum} bnum={bnum} />;
+  const {
+    value,
+  }: {
+    value: BodyItem['type'];
+  } = input;
+  if (value === 'HTML') {
+    return (
+      <Field
+        name={memberName}
+        component={EditHTMLField}
+        placeholder="Body item..."
+      />
+    );
+  }
+  let body;
+  switch (value) {
     case 'Code':
-      return <Code info={body} qnum={qnum} pnum={pnum} bnum={bnum} />;
+      body = <Code qnum={qnum} pnum={pnum} bnum={bnum} />;
+      break;
     case 'AllThatApply':
-      return <AllThatApply info={body} qnum={qnum} pnum={pnum} bnum={bnum} />;
+      body = <AllThatApply qnum={qnum} pnum={pnum} bnum={bnum} />;
+      break;
     case 'CodeTag':
-      return <CodeTag info={body} qnum={qnum} pnum={pnum} bnum={bnum} />;
+      body = <CodeTag qnum={qnum} pnum={pnum} bnum={bnum} />;
+      break;
     case 'YesNo':
-      return <YesNoInput info={body} qnum={qnum} pnum={pnum} bnum={bnum} />;
+      body = <YesNo qnum={qnum} pnum={pnum} bnum={bnum} />;
+      break;
     case 'MultipleChoice':
-      return <MultipleChoice info={body} qnum={qnum} pnum={pnum} bnum={bnum} />;
+      body = <MultipleChoice qnum={qnum} pnum={pnum} bnum={bnum} />;
+      break;
     case 'Text':
-      return <Text info={body} qnum={qnum} pnum={pnum} bnum={bnum} />;
+      body = <Text qnum={qnum} pnum={pnum} bnum={bnum} />;
+      break;
     case 'Matching':
-      return <Matching info={body} qnum={qnum} pnum={pnum} bnum={bnum} />;
+      body = <Matching qnum={qnum} pnum={pnum} bnum={bnum} />;
+      break;
     default:
-      throw new ExhaustiveSwitchError(body);
+      throw new ExhaustiveSwitchError(value);
   }
+  return (
+    <FormSection name={memberName}>
+      {body}
+    </FormSection>
+  );
 };
 
-const WrappedBody: React.FC<BodyProps> = (props) => {
+export interface BodyProps {
+  memberName: string;
```

Comment:
This is so that the bodyitem knows its address into the store. This way we can either directly render the HTML for editing, or render a FormSection that is already prefixed to the current BodyItem.